components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/listener/IdentityStoreEventListener.java

```diff
@@ -126,7 +127,8 @@ public boolean doPreAddUser(String userName,
 
             Map.Entry<String, String> claim = it.next();
             if (claim.getKey().contains(UserCoreConstants.ClaimTypeURIs.IDENTITY_CLAIM_URI_PREFIX)
-                    && !(identityDataStore instanceof UserStoreBasedIdentityDataStore)) {
+                    && !(identityDataStore instanceof UserStoreBasedIdentityDataStore ||
+                    isStoreIdentityClaimsInUserStore(userStoreManager))) {
```

Comment:
IMO this method name does not sound right. `isStoreIdentityClaimsInUserStoreEnabled` WDYT? Or any other name?

Reply:
Addressed with https://github.com/wso2-extensions/identity-governance/pull/648/commits/a656aec88d3a39cb538298c6bbe27cc90de4e8c8